src/toolkits/object_detection/object_detector.cpp

```diff
@@ -350,6 +350,74 @@ void object_detector::load_version(iarchive& iarc, size_t version) {
   _load_version(iarc, version, *nn_spec_, state, anchor_boxes());
 }
 
+void object_detector::import_from_custom_model(variant_map_type model_data,
+                                               size_t version) {
+  auto it = model_data.find("_model");
+  flex_dict model = variant_get_value<flex_dict>(it->second);
+  model_data.erase(it);
+  auto it2 = model_data.find("_grid_shape");
+  std::vector<size_t> shape =
+      variant_get_value<std::vector<size_t>>(it2->second);
```

Comment:
Just in case, let's make this robust to `_grid_shape` not being present in the dictionary